init.lua

```diff
@@ -1,54 +1,205 @@
+			bits = bits .. "1"
+			remaining = remaining - pow
+		else
+			bits = bits .. "0"
+		end
+	end
+	return bits
+end
+local function fromBinaryString(bits)
+	return tonumber(bits, 2)
+end
+local function decodeBase64(encoded)
+	local bitstr = ""
+	local decoded = ""
+	-- decode chars into bitstring
+	for i = 1, string.len(encoded) do
+		local offset, _ = string.find(alpha, string.sub(encoded, i, i))
+		if offset == nil then
+			error("Bad base64 character " .. string.sub(encoded, i, i))
+		end
+		bitstr = bitstr .. toBinaryString(offset - 1)
+	end
+	-- decode bitstring back to chars
+	for i = 1, string.len(bitstr), 8 do
+		decoded = decoded .. string.char(fromBinaryString(string.sub(bitstr, i, i + 7)))
+	end
+	return decoded
+end
+-- json handling
+local json = {}
 
-function M:peek(job)
+-- Returns pos, did_find; there are two cases:
+-- 1. Delimiter found: pos = pos after leading space + delim; did_find = true.
+-- 2. Delimiter not found: pos = pos after leading space;     did_find = false.
+-- This throws an error if err_if_missing is true and the delim is not found.
+local function skip_delim(str, pos, delim, err_if_missing)
+	pos = pos + #str:match("^%s*", pos)
+	if str:sub(pos, pos) ~= delim then
+		if err_if_missing then
+			error("Expected " .. delim .. " near position " .. pos)
+		end
+		return pos, false
+	end
+	return pos + 1, true
+end
 
-	local child = Command("jwt-decoder")
-		:args({
-			"-f",
-			tostring(job.file.url),
-		})
-		:stdout(Command.PIPED)
-		:stderr(Command.PIPED)
-		:spawn()
-
-	if not child then
-		return require("code").peek(job)
-	end
-
-	local limit = job.area.h
-	local i, lines = 0, ""
-	repeat
-		local next, event = child:read_line()
-		if event == 1 then
-			return require("code").peek(job)
-		elseif event ~= 0 then
-			break
+-- Expects the given pos to be the first character after the opening quote.
+-- Returns val, pos; the returned pos is after the closing quote character.
+local function parse_str_val(str, pos, val)
+	val = val or ""
+	local early_end_error = "End of input found while parsing string."
+	if pos > #str then
+		error(early_end_error)
+	end
+	local c = str:sub(pos, pos)
+	if c == '"' then
+		return val, pos + 1
+	end
+	if c ~= "\\" then
+		return parse_str_val(str, pos + 1, val .. c)
+	end
+	-- We must have a \ character.
+	local esc_map = { b = "\b", f = "\f", n = "\n", r = "\r", t = "\t" }
+	local nextc = str:sub(pos + 1, pos + 1)
+	if not nextc then
+		error(early_end_error)
+	end
+	return parse_str_val(str, pos + 2, val .. (esc_map[nextc] or nextc))
+end
+
+-- Returns val, pos; the returned pos is after the number's final character.
+local function parse_num_val(str, pos)
+	local num_str = str:match("^-?%d+%.?%d*[eE]?[+-]?%d*", pos)
+	local val = tonumber(num_str)
+	if not val then
+		error("Error parsing number at position " .. pos .. ".")
+	end
+	return val, pos + #num_str
+end
+
+json.null = {} -- one-off table to represent the null value.
+function json.parse(str, pos, end_delim)
```

Comment:
I don't see why have to parse the JSON, YAZI has a built-in JSON tool that we could forward the resulting string to once it has been decoded from JWT into proper JSON (identified by leading curly brace).